dali/test/python/test_external_source_dali.py

```diff
@@ -741,3 +744,36 @@ def pipeline():
 def test_repeat_last_var_batch():
     for device in ['cpu', 'gpu']:
         yield _check_repeat_last_var_batch, device
+
+
+def _check_blocking(device):
+    batch_size = 5
+    prefetch_queue_depth = 10
+
+    @pipeline_def
+    def test_pipeline():
+        jpegs = fn.external_source( dtype=types.UINT8, name="test_source", blocking=True)
+        decoded = fn.decoders.image(jpegs, device=device)
+        return decoded
+    root = os.path.join(DALI_EXTRA_PATH, 'db', 'single', 'jpeg')
+    file = glob.glob("**/*.jpg", root_dir=root)[0]
+    file = os.path.join(root, file)
+    data_to_feed = [np.fromfile(file, dtype=np.uint8) for _ in range(batch_size)]
```

Comment:
I'd suggest using a much simpler (and faster) test - just feed a bunch of fixed numpy arrays, return them unchanged and compare.

Reply:
Fixed